### DISCUSSION TITLE
question about cpxInitRoute

### DISCUSSION THREAD

**Discussion by noahzn** [bitcraze/discussions#820] [Q&A]
Hi, I need to capture images and send them to my laptop via WIFI. Meanwhile, I also send messages from GAP8 to the APP layer. So, I have some problems with the `cpxInitRoute`.

My code is based on the face detection demo, and I use: `cpxInit();    cpxEnableFunction(CPX_F_WIFI_CTRL);` in the beginning of the function `void start_example(void)`. Then, I set `cpxInitRoute(CPX_T_GAP8, CPX_T_WIFI_HOST, CPX_F_APP, &txp.route)` in the face detection task. In this way, my opencv viewer can receive images sent by GAP8, but the GAP8->APP layer doesn't work. 

If I set `cpxInitRoute(CPX_T_GAP8, CPX_T_STM32, CPX_F_APP, &txp2.route)`, then the sending function works (GAP8->APP), but the opencv viewer doesn't work.

How can I make them work together? How should I set the `cpxInitRoute` in two tasks? Thank you.

**Comment by noahzn** [comment]
It seems that the problem can be solved by setting a **long** delay to the second task (sending messages from GAP8 to the APP layer). I need to wait until the UART is open.

**Reply by ataffanel** [reply]
Thanks for the update and for a potential fix. Everyone that has knowledge about CPX is currently on Vacation so it is good that you managed to solve it :) (I was about to have a look but it would have taken me a long time to get setup). This sounds like a bug we should fix (at the very least with doc, but even better on the API). If you have a good way to reproduce the bug it would be great if you could make an issue on the [crazyflie-firmware](https://github.com/bitcraze/crazyflie-firmware/issues) repos.
